### PR TITLE
feat(E2EEncryption): File encryption support

### PR DESCRIPTION
**Pull request by rodrigok**
## Issue(s)

Task: [E2EE2-4](https://rocketchat.atlassian.net/browse/E2EE2-4)

## Further comments

- [x] Move `msg` to inside content
- [x] Hack to be able to download files on chrome
- [x] Decide if the APIs will be under room domain
- [x] Decide what to show when e2ee is disabled (check skipped test)
- [x] Cronjob to remove temporary files older than X hours
- [x] Mark uploaded files as temporary
- [x] Mark as non temporary when sending the message
- [x] Anonimize file name
- [x] Decide if `content` will be string and data
- [x] Decide if the key info will be inside the attachment info
- [x] Should simplify the data structure and make it compatible on read?
- [x] Deprecate old upload API
- [ ] ~Prepare API to accept multiple files on media confirm?~
- [ ] Tests
  - [x]  Old format message
  - [ ] New Upload API
    - [x] Current old API tests
    - [ ] Media Confirm access only
    - [ ] Not confirmed media access
    - [ ] Auto removal of not confirmed media?
- [ ] Update Docs
- [x] Client generated data:
  - [ ] ~Thumbs?~
  - [ ] ~Blurred background?~
  - [x] Extra file info? Width, Height

[E2EE2-4]: https://rocketchat.atlassian.net/browse/E2EE2-4?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/client/components/message/variants/room/RoomMessageContent.tsx [hunk 1]
```diff
@@ -46,6 +55,31 @@ const RoomMessageContent = ({ message, unread, all, mention, searchText }: RoomM
 	const normalizedMessage = useNormalizedMessage(message);
 	const isMessageEncrypted = encrypted && normalizedMessage?.e2e === 'pending';
 
+	if (encrypted && normalizedMessage?.attachments?.length) {
+		normalizedMessage.attachments.forEach((attachment) => {
+			if (!attachment.encryption) {
+				return;
+			}
+
+			const key = Base64.encode(JSON.stringify(attachment.encryption));
```
Comment (diegolmello) [inline apps/meteor/client/components/message/variants/room/RoomMessageContent.tsx:64]: I'm able to download files without appending key as param. Is that expected?

File: apps/meteor/app/e2e/client/rocketchat.e2e.room.js [hunk 2]
```diff
@@ -333,69 +334,72 @@ export class E2ERoom extends Emitter {
...
-			return;
-		}
+		// if (!this.isSupportedRoomType(this.typeOfRoom)) {
+		// 	return;
+		// }
 
 		const fileArrayBuffer = await readFileAsArrayBuffer(file);
 
 		const vector = crypto.getRandomValues(new Uint8Array(16));
+		const key = await generateAESCTRKey();
```
Comment (diegolmello) [inline apps/meteor/app/e2e/client/rocketchat.e2e.room.js:353]: Can't we use room key?

Comment (rodrigok) [reply apps/meteor/app/e2e/client/rocketchat.e2e.room.js:353]: @diegolmello No, the room's key is CBC, I've tried and it's not compatible.

**Review by diegolmello — COMMENTED** [review]
Encryption is not working in threads

File: apps/meteor/client/views/room/contextualBar/RoomFiles/components/FileItem.tsx [hunk 3]
```diff
@@ -22,12 +24,28 @@ type FileItemProps = {
 
 const FileItem = ({ fileData, onClickDelete }: FileItemProps) => {
 	const format = useFormatDateAndTime();
-	const { _id, name, url, uploadedAt, type, typeGroup, user } = fileData;
+	const [file, setFile] = useState<IUploadWithUser>(fileData);
+	const { _id, path, name, uploadedAt, type, typeGroup, user } = file;
+	console.log(123, file);
```
Comment (diegolmello) [inline apps/meteor/client/views/room/contextualBar/RoomFiles/components/FileItem.tsx]: 👀

**Review by ggazzo — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

**Pull request merged**